### ISSUE TITLE
Feature request: ESPHome integration

### ISSUE THREAD

**Issue by timotoots** [pschatzmann/arduino-audio-tools#118]
Hello Phil. First of all thanks for the great initiative for this library. Seems perfect for all kinds of crazy stuff.

Would this library be feasible to use as a ESPHome custom component? That project is in big need for audio player component and it would get a lot of use as home music players etc.

Some discussion in here:
https://github.com/esphome/feature-requests/issues/599

**Comment by pschatzmann** [comment]
I think you can use my different libraries together with ESPHome to build some cool stuff. E.g. with my library [arduino-simple-tts](https://github.com/pschatzmann/arduino-simple-tts) you could play some sound recordings based on keywords...

**Comment by jesserockz** [comment]
We would love to pull something like this into ESPHome.

The main issue I see though is that it has a reliance on Arduino (and it's provided libraries) which raises flags for us as we are trying to move ESP32 fully to ESP-IDF.

**Comment by pschatzmann** [comment]
Well on my side it was a conscious decision to make this library dependent on Arduino because of the Arduino Streams, so you can deal with (audio) data via Serial, UDP, TCP, Files, Bluetooth...with just one interface. To do this in ESP-IDF would be a big nightmare...

**Comment by timotoots** [comment]
@jesserockz If the way forward with ESPHome is with ESP-IDF then I guess it could be possible to include ESP-ADF as well? It provides a lot of audio streams functionality as well.

**Comment by pschatzmann** [comment]
I agree - that would be the logical choice. It's just not as easy to use ...

**Comment by pschatzmann** [comment]
Moved to discussion - ideas
